scipy/linalg/_decomp_svd.py

```diff
@@ -299,6 +303,101 @@ def diagsvd(s, M, N):
         raise ValueError("Length of s must be M or N.")
 
 
+def higher_order_svd(
+    a: npt.ArrayLike, *, compact_rtol: float | None = None, check_finite: bool = True
+) -> tuple[list[np.ndarray], np.ndarray]:
+    """Higher-order SVD (HOSVD)
+
+    Factorizes the M-D tensor `a` into a list of M matrices ``U_k`` containing
+    the left singular vectors of the unfolded tensor with respect to the
```

Comment:
Have to admit I don't know off the the cuff what an unfolded tensor is.

Googling around gives https://jeankossaifi.com/blog/unfolding.html (which references `"Tensor Decompositions and Applications"in SIAM REVIEW, 2009)`) and https://docs.pytorch.org/docs/stable/generated/torch.nn.Unfold.html

Are these the same as here? If so, it'd be best to add a reference and give a definition, if possible. My first knee-jerk reaction was can we have a math-y definition in the Einstei tensor notation. If that's difficult, can we please have _some_ explanation, a code example, and a link to the literature.

Reply:
Don't know what to do here